platform/darwin/src/MGLStyle.mm

```diff
@@ -104,6 +110,33 @@ + (NSURL *)emeraldStyleURL {
     return styleLayer;
 }
 
+- (MGLSource *)sourceWithIdentifier:(NSString *)identifier
+{
+    auto s = self.mapView.mbglMap->getSource(identifier.UTF8String);
+    
+    Class clazz = [self classFromSource:s];
+    
+    MGLSource *source = [[clazz alloc] init];
+    source.sourceIdentifier = identifier;
+    source.source = s;
+    
+    return source;
+}
+
+- (Class)classFromSource:(mbgl::style::Source *)source
+{
+    if (dynamic_cast<mbgl::style::VectorSource *>(source)) {
```

Comment:
Use `source->is<mbgl::style::VectorSource>`, which relies on a generic interface, instead of `dynamic_cast`, which relies on RTTI.